File: src/data/models/UtilityPriceHistory.js
import DataType from 'sequelize';
import Model from '../sequelize';

const UtilityPriceHistory = Model.define('UtilityPriceHistory', {
  id: {
    type: DataType.INTEGER,
    allowNull: false,
    primaryKey: true,
    autoIncrement: true,
  },
  name: {
    type: DataType.STRING,
    allowNull: false,
  },
  uniqueIdentifier: {
    type: DataType.STRING(25),
    allowNull: false,
  },
  utilityState: {
    type: DataType.STRING(2),
    allowNull: false,
  },
  utilityType: {
    type: DataType.STRING(255),
    allowNull: false,
  },
  pricePerUnit: {
    type: DataType.FLOAT,
    allowNull: true,
  },
  priceUnit: {
    type: DataType.STRING(50),
    allowNull: true,
  },
  flatRate: {
    type: DataType.STRING(120),
    allowNull: true,
  },
  priceStructure: {
    type: DataType.STRING(255),
    allowNull: false,
  },
  hasCancellationFee: {
    type: DataType.BOOLEAN,
    allowNull: false,
    defaultValue: false,
  },
  cancellationFee: DataType.STRING(255),
  hasMonthlyFee: {
    type: DataType.BOOLEAN,
    allowNull: false,
    defaultValue: false,
  },
  monthlyFee: DataType.STRING(255),
  hasEnrollmentFee: {
    type: DataType.BOOLEAN,
    allowNull: false,
    defaultValue: false,
  },
  enrollmentFee: DataType.STRING(255),
  hasNetMetering: {
    type: DataType.BOOLEAN,
    allowNull: false,
    defaultValue: false,
  },
  requiresDeposit: {
    type: DataType.BOOLEAN,
    allowNull: false,
    defaultValue: false,
  },
  hasBulkDiscounts: {
    type: DataType.BOOLEAN,
    allowNull: false,
    defaultValue: false,
  },
  isIntroductoryPrice: {
    type: DataType.BOOLEAN,
    allowNull: false,
    defaultValue: false,
  },
  hasRenewable: {
    type: DataType.BOOLEAN,
    allowNull: false,
    defaultValue: false,
  },
  renewablePercentage: DataType.FLOAT,
  termMonthLength: DataType.INTEGER,
  termEndDate: DataType.DATE,
  offerUrl: DataType.STRING(256),
  supplierAddress1: DataType.STRING,
  supplierAddress2: DataType.STRING,
  supplierCity: DataType.STRING,
  supplierState: DataType.STRING,
  supplierZip: DataType.STRING,
  supplierPhone: DataType.STRING,
  comments: DataType.STRING,
  createdAt: {
    type: DataType.DATE,
    allowNull: false,
    defaultValue: DataType.NOW,
  },
});

export default UtilityPriceHistory;

File: src/CompareHare.Domain/Entities/UtilityPriceHistory.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using CompareHare.Domain.Services;

namespace CompareHare.Domain.Entities
{
    public class UtilityPriceHistory : ICreatedDateTimeTracker, IModifiedDateTimeTracker
    {
        public int Id { get; set; }

        public int StateUtilityIndexId { get; set; }
        public StateUtilityIndex StateUtilityIndex { get; set; }

        [Required, MaxLength(256)]
        public string Name { get; set; }

        [MaxLength(50)]
        public string PriceUnit { get; set; }

        public float? PricePerUnit { get; set; }

        [MaxLength(120)]
        public string FlatRate { get; set; }

        [MaxLength(255)]
        public string PriceStructure { get; set; }

        public bool? HasCancellationFee { get; set; }
        public string CancellationFee { get; set; }

        public bool? HasMonthlyFee { get; set; }
        public string MonthlyFee { get; set; }

        public bool? HasEnrollmentFee { get; set; }
        public string EnrollmentFee { get; set; }

        public bool? HasNetMetering { get; set; }
        public string NetMetering { get; set; }

        public bool? RequiresDeposit { get; set; }
        public bool? HasBulkDiscounts { get; set; }
        public bool? IsIntroductoryPrice { get; set; }

        public bool? HasRenewable { get; set; }
        public float? RenewablePercentage { get; set; }

        public int? TermMonthLength { get; set; }

        [MaxLength(40)]
        public string SupplierPhone { get; set; }

        public bool? HasTermEndDate { get; set; }
        public DateTime? TermEndDate { get; set; }

        [MaxLength(64)]
        public string OfferId { get; set; }

        [MaxLength(512)]
        public string OfferUrl { get; set; }

        [MaxLength(512)]
        public string Comments { get; set; }

        public IEnumerable<AlertMatchUtilityPriceHistory> Alerts { get; set; }

        public DateTimeOffset CreatedDate { get; set; }
        public DateTimeOffset? ModifiedDate { get; set; }
    }
}
